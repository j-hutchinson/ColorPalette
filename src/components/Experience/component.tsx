import React from 'react';
import styled from 'styled-components';
import { CKODesc, ElsDesc, CFGDesc } from '../../css/copy';
import { SectionStyles } from '../../css/tokens';
import JobProfile from '../JobProfile/component';

const StyledContainer = styled.div`
  margin-top: -150px;

  @media screen and (max-width: 800px) {
    margin-top: 0px;
  }
`;

const StyledWrapper = styled.section`
  ${SectionStyles};
`;

const StyledHeading = styled.h2`
  font-size: 36px;
`;

const StyledList = styled.ul`
  list-style: none;
  padding: 0;
`;

const StyledListItem = styled.li`
  :last-child {
    border: none
  }
`;

const Experience = (): JSX.Element => (
  <StyledContainer>
    <StyledWrapper >
      <StyledHeading>Where have I been? 🙇‍♂️</StyledHeading>
      <StyledList>
        <StyledListItem>
          <JobProfile name="Checkout.com" startDate="04/01/21" title="Senior Software Engineer" desc={CKODesc} />
        </StyledListItem>
        <StyledListItem>
          <JobProfile name="Code First Girls" startDate="01/20" endDate="04/20" title="Instructor" desc={CFGDesc} />
        </StyledListItem>
        <StyledListItem>
          <JobProfile name="Elsevier" startDate="04/09/17" title="Graduate Engineer -> Frontend Engineer" desc={ElsDesc} />
        </StyledListItem>
      </StyledList>
    </StyledWrapper >
  </StyledContainer>
);

export default Experience;
