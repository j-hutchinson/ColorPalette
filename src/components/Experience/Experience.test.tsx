import { shallow } from 'enzyme';
import React from 'react';
import Experience from './component';

describe('Experience component', () => {
    test('component matches snapshot', () => {
        expect.assertions(1);

        const wrapper = shallow(<Experience />);

        expect(wrapper).toMatchSnapshot();
    });
});