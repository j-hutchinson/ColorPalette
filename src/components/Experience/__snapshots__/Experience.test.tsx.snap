// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Experience component component matches snapshot 1`] = `
<styled.section>
  <styled.h2>
    Where have I been? 🙇‍♂️
  </styled.h2>
  <styled.ul>
    <styled.li>
      <JobProfile
        desc="Software developer specialising in front end web development utilising JavaScript, React, GraphQL, Docker and AWS. Work in a cross-functional team gaining exposure to the entire product lifecycle from design sessions and user testing to making product decisions in addition to autonomously implementing business features."
        name="Checkout.com"
        startDate="04/01/21"
        title="Senior Software Engineer"
      />
    </styled.li>
    <styled.li>
      <JobProfile
        desc="Leading and designing the Front-end architecture within a cross-functional team developing a React web app and GraphQL service supporting multiple teams. Responsible for multiple micro-frontends supporting mendeley.com"
        name="Elsevier"
        startDate="04/09/17"
        title="Graduate Engineer -> Frontend Engineer"
      />
    </styled.li>
  </styled.ul>
</styled.section>
`;
